python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -563,7 +569,10 @@ def agg(self, func):
                 ):
                     # Structs lose their labels which we reconstruct here
                     col = col._with_type_metadata(cudf.ListDtype(orig_dtype))
-                data[key] = col
+                if is_empty and is_reduction and len(col) == 0:
```

Comment:
If the values of `is_empty` and `is_reduction` are only used here, maybe we can declare/evaluate them closer to here?

Reply:
Yupp, moved it closer.